### PR TITLE
[IMP] mail, mail_group: simplify kanban archs

### PR DESCRIPTION
**Pull request by laro-odoo**
In this commit we have simplified the kanban arch for the mail and mail_group modules.  The goal is to simplify them, make them easier to read, and use bootstrap utility classnames.
- Previously, we used `kanban-box`, but now we are using `kanban-card` instead.
- Deprecated `oe_kanban_global_click` and `oe_kanban_global_click_edit`.
- More use of `<field/>` tags
- Removed the `oe_kanban_colorpicker` class and replaced it with the `kanban_color_picker` widget.
- Changed type='edit' to type='open' to open records. since version 16, records always open in edit mode by default.
- `kanban_image` from rendering context, is deprecated so we use `<field name=... widget=image/>` instead

Task-3992107

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/views/discuss_channel_views.xml [hunk 1]
```diff
@@ -8,35 +8,25 @@
             <field name="priority" eval="10"/>
             <field name="arch" type="xml">
                 <kanban>
-                    <field name="id"/>
-                    <field name="description"/>
                     <field name="is_member"/>
                     <field name="group_ids"/>
                     <field name="active"/>
                     <templates>
                         <t t-name="kanban-description">
```
Comment (aab-odoo) [inline addons/mail/views/discuss_channel_views.xml:15]: this is dead code?? Where is this template called?

File: addons/mail/views/mail_activity_views.xml [hunk 2]
```diff
@@ -108,28 +108,21 @@
...
+                            <i t-if="record.icon.value" t-attf-class="fa #{record.icon.value} fa-fw"
+                               role="img" aria-label="Activity Type Name" title="Activity Type Name"/>
+                            <field name="name"/>
+                        </div>
+                        <field t-if="record.res_model.value" name="res_model"/>
+                        <div t-if="record.summary.raw_value">
+                            Default Summary: <field name="summary"/>
                         </div>
+                        <footer>
+                            <field class="float-end" name="default_user_id"
```
Comment (aab-odoo) [inline addons/mail/views/mail_activity_views.xml:123]: prefer ms-auto then float-end?

File: addons/mail/static/src/scss/kanban_view.scss [hunk 3]
```diff
@@ -14,18 +14,8 @@ $o-kanban-attachement-image-size: 80px;
         .o_kanban_image {
             width: $o-kanban-attachement-image-size;
 
-            + div {
```
Comment (aab-odoo) [inline addons/mail/static/src/scss/kanban_view.scss:17]: comment for above: class o_kanban_attachment is always set on the o_kanban_record, so selector line 11 is useless

File: addons/mail/views/mail_message_views.xml [hunk 4]
```diff
@@ -139,44 +139,31 @@
...
-                            <div class="oe_kanban_global_area oe_kanban_global_click o_kanban_attachment">
-                                <div class="o_kanban_image">
-                                    <div class="o_kanban_image_wrapper">
-                                        <t t-set="webimage" t-value="new RegExp('image.*(gif|jpeg|jpg|png|webp)').test(record.mimetype.value)"/>
-                                        <div t-if="record.type.raw_value == 'url'" class="o_url_image fa fa-link fa-3x text-muted" aria-label="Image is a link"/>
-                                        <img t-elif="webimage" t-attf-src="/web/image/#{record.id.raw_value}" width="100" height="100" alt="Document" class="o_attachment_image"/>
-                                        <div t-else="!webimage" class="o_image o_image_thumbnail" t-att-data-mimetype="record.mimetype.value"/>
-                                    </div>
+                        <t t-name="kanban-card" class="o_kanban_attachment flex-row">
+                            <div class="o_kanban_image">
```
Comment (aab-odoo) [inline addons/mail/views/mail_message_views.xml:148]: isn't this a classical aside/main layout? Why do we need those custom (that previously where standard) classnames (o_kanban_image, o_kanban_details, o_kanban_details_wrapper) ?

File: addons/mail_group/views/mail_group_views.xml [hunk 5]
```diff
@@ -17,32 +17,29 @@
...
+                        </aside>
+                        <main>
+                            <span class="fw-bold fs-5">#<field name="name"/></span>
+                            <field class="text-truncate text-nowrap" name="description"/>
+                            <span>
+                                <field name="member_count"/>
+                                <t t-if="record.member_count.raw_value > 1">Members</t>
+                                <t t-else="">Member</t>
+                            </span>
+                            <field name="is_member" invisible="1"/>
```
Comment (aab-odoo) [inline addons/mail_group/views/mail_group_views.xml:39]: not necessary, you already define it outside the template

File: addons/mail/views/mail_message_views.xml (hunk 4 above)
Comment (laro-odoo) [reply addons/mail/views/mail_message_views.xml:148]: As you can observe in the `kanban_view.scss`, the height and width of the kanban are primarily determined by `$o-kanban-attachment-image-size`, which corresponds to the largest attachment uploaded. If we don't keep these classes, the height of the kanban will vary across different rows. Therefore, we need to use the CSS classes to maintain consistent dimensions.

File: addons/mail/views/discuss_channel_views.xml (hunk 1 above)
Comment (laro-odoo) [reply addons/mail/views/discuss_channel_views.xml:15]: I believe it is indeed dead code now. I was also confused about this, as I couldn't find its implementation anywhere else in Odoo.

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**